plugins/repository-s3/src/test/java/org/elasticsearch/cloud/aws/AwsS3ServiceImplTests.java

```diff
@@ -398,84 +324,84 @@ private static Settings generateRepositorySettings(String key, String secret, St
             "s3.eu-central-1.amazonaws.com");
-        launchAWSEndpointTest(generateRepositorySettings("repository_key", "repository_secret", null, "repository.endpoint", null),
+        assertEndpoint(generateRepositorySettings("repository_key", "repository_secret", null, "repository.endpoint", null),
             settings, "repository.endpoint");
+        assertWarnings("[" + InternalAwsS3Service.LEGACY_REGION_SETTING.getKey() + "] setting was deprecated",
+                       "[" + S3Repository.Repository.REGION_SETTING.getKey() + "] setting was deprecated",
+                       "[" + InternalAwsS3Service.CLOUD_S3.REGION_SETTING.getKey() + "] setting was deprecated",
+                       "[" + S3Repository.Repository.ENDPOINT_SETTING.getKey() + "] setting was deprecated");
     }
 
     public void testInvalidRegion() {
         Settings settings = Settings.builder()
-            .put(InternalAwsS3Service.REGION_SETTING.getKey(), "does-not-exist")
+            .put("aws.config.default.region", "does-not-exist")
             .build();
         IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> {
-            launchAWSEndpointTest(generateRepositorySettings("repository_key", "repository_secret", null, null, null), settings, null);
+            assertEndpoint(generateRepositorySettings("repository_key", "repository_secret", null, null, null), settings, null);
         });
         assertThat(e.getMessage(), containsString("No automatic endpoint could be derived from region"));
-
-        launchAWSEndpointTest(generateRepositorySettings("repository_key", "repository_secret", "eu-central", null, null), settings,
-            "s3.eu-central-1.amazonaws.com");
-        launchAWSEndpointTest(generateRepositorySettings("repository_key", "repository_secret", null, "repository.endpoint", null),
-            settings, "repository.endpoint");
     }
 
-    protected void launchAWSEndpointTest(Settings singleRepositorySettings, Settings settings,
-                                         String expectedEndpoint) {
-        String region = S3Repository.getValue(singleRepositorySettings, settings,
-            S3Repository.Repository.REGION_SETTING, S3Repository.Repositories.REGION_SETTING);
-        String endpoint = S3Repository.getValue(singleRepositorySettings, settings,
-            S3Repository.Repository.ENDPOINT_SETTING, S3Repository.Repositories.ENDPOINT_SETTING);
-
-        String foundEndpoint = InternalAwsS3Service.findEndpoint(logger, settings, endpoint, region);
+    private void assertEndpoint(Settings repositorySettings, Settings settings,
+                                  String expectedEndpoint) {
+        String configName = InternalAwsS3Service.CONFIG_NAME.get(repositorySettings);
+        String foundEndpoint = InternalAwsS3Service.findEndpoint(logger, repositorySettings, settings, configName);
         assertThat(foundEndpoint, is(expectedEndpoint));
     }
```

Comment:
Adding a comment unrelated to your change but `generateRepositorySettings()` method always gets `null` for the `maxRetries` parameter. May be we can remove it? Or really test it?